src/app/api/cms/messages/get_message.ts

```diff
@@ -1,68 +1,56 @@
 import * as env from "@utils/config.js";
-import fs from "fs";
-import { Builder, parseString } from "xml2js";
+import { NextResponse, type NextRequest } from "next/server";
 
 const api_url: string | undefined = process.env.ACTIVEITS_API_HOSTNAME;
 const api_port: string | undefined = process.env.ACTIVEITS_API_PORT;
 
-export default async function get_message(req: Request, res: Response) {
+export default async function get_message(req: NextRequest, res: NextResponse) {
+  const dict_response: Record<string, string> = {};
   try {
-    const active_client: ActiveITSClient = new ActiveITSClient(
-      api_url!,
-      Number(api_port)
-    );
-
-    // Connect to the client
+    const active_client: ActiveITSClient = new ActiveITSClient(api_url!, Number(api_port));
+    const db_client: ActiveITSDBClient = ActiveITSDBClient.getInstance();
+    db_client.initialize();
+    const searchParams = req.nextUrl.searchParams;
+    const goldeneye_username: string | null = searchParams.get("goldeneye_username");
+    const data = searchParams.get("data"); //need to verify data type
     await active_client.connect();
-    console.log("Connected successfully");
-
-    // Read XML data
-    const xmlData = fs.readFileSync(env.mas_login_filepath, "utf-8");
-
-    // Parse the XML data
-    const result = await parseXmlAsync(xmlData);
-
-    // Modify the parsed data
-    const authenticateReq = result.authenticateReq;
-
-    if (!authenticateReq) {
-      throw new Error("XML structure is invalid.");
-    }
-
-    authenticateReq.refId[0] = "testuser"; // Need to change
-    authenticateReq.username[0] = "testuser"; // Need to change
-    active_client.getPassword("testuser").then((password: string) => {
-      //need to update
-      if (password != "") {
-        authenticateReq.password[0] = "password";
-        console.log("Password:", password);
-      } else {
-        console.log("User not found or password is null/undefined.");
+    await db_client.connect();
+    try {
+      const dict_cms_num_to_activeITS_num: Promise<"" | Record<string, string>> | any = db_client.get_cms_id_to_ritms_id_dict();
+      const mas_security_token: string = await active_client.login_to_subsystem(env.DATABUS_SUBSYSTEM, goldeneye_username, db_client);
+      const decoded_cml_queues: string = await active_client.get_queue_msgs(mas_security_token, String(goldeneye_username));
+      active_client.close();
+      const cms_num_to_current_msg: Promise<"" | Record<string, string>> | any = await db_client.get_current_cms_msgs();
+      const RECEIVED_DATA: string = data ? String(data) : "";
+      const parsedData = JSON.parse(RECEIVED_DATA);
+      let current_msg = "";
+      for (const cms_id in parsedData) {
+        const key: string = String(cms_id);
+        current_msg = cms_num_to_current_msg[key];
+        active_client
+          .is_travel_time_message(parseInt(dict_cms_num_to_activeITS_num[cms_id]), decoded_cml_queues, current_msg)
```

Comment:
You don't need this Promise chaining with `.then` and `.catch` since `active_client.is_travel_time_message` should be a synchronous function. Instead, you could do
```javascript
try {
  const result = active_client.is_travel_time_message(...)
} catch (error) {
  console.error(error);
}
```

Reply:
Hi Jared, 
I am returning a promise from that function ,so I need to handle that promise . But I did see I did not check result for handling the current msg parameter. 
I will be updating that .